lib/terraform.go

```diff
@@ -1,19 +1,20 @@
 package terraspec
 
 import (
-	"path"
-	"path/filepath"
-
+	"github.com/hashicorp/terraform/addrs"
 	"github.com/hashicorp/terraform/configs/configload"
 	"github.com/hashicorp/terraform/providers"
 	"github.com/hashicorp/terraform/terraform"
 	"github.com/hashicorp/terraform/tfdiags"
 	"github.com/zclconf/go-cty/cty"
+	"log"
+	"path"
+	"path/filepath"
 )
 
 // NewContext creates a new terraform.Context able to compute configs in the context of terraspec
 // It returns the built Context or a Diagnostics if error occured
-func NewContext(dir, varFile string, resolver providers.Resolver) (*terraform.Context, tfdiags.Diagnostics) {
+func NewContext(dir, varFile string, resolver ProviderResolver, mockMetadata *MockMetadata) (*terraform.Context, tfdiags.Diagnostics) {
```

Comment:
so I added the metadata mock through here. Its the best I could think of. The change of the resolver is terraform changing the contract.